### PR TITLE
docs: created new documentation for Kasten Backup and Restore using Replicated PV Mayastor Snapshots - FileSystem

### PR DESCRIPTION
**Pull request by balaharish7**

### CODE REVIEW

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
+---
+id: kastenbrfs
+title: Kasten Backup and Restore using Replicated PV Mayastor Snapshots - FileSystem
+keywords: 
+  - Kasten Backup and Restore using Replicated PV Mayastor Snapshots - FileSystem
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:5]: The keywords, titles and description needs to include OpenEBS for the search engines to source these documents. Same for Velero guides as well.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+In this example we have changed the 'svc' type to **NodePort**:
+
+1. Forward a local port to the Kasten ingress port.
+
+```
+kubectl --namespace kasten-io port-forward service/gateway 8080:80
+```
+
+2. Change the 'svc' type as **NodePort**.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:108]: @datacore-tilangovan do we need this even after port-forwarding?

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+#### Backup
+
+Once the policies have been created, it is possible to run the backup. In this scenario, we have created policies to run “on-demand”. A snapshot can be scheduled based on the available options. Example: hourly/weekly
+
+- Select **Run Once > Yes, Continue** to trigger the snapshot. Backup operations convert application and volume snapshots into backups by transforming them into an infrastructure-independent format and then storing them in a target location (Google Cloud Bucket).
+
+![run-once](../../assets/run-once.png)
+
+The snapshots status can be monitored and exported from the Dashboard. The backup had been successfully completed and exported to the storage location.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:248]: "status can be monitored and exported" how can status be exported?

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+Once the policies have been created, it is possible to run the backup. In this scenario, we have created policies to run “on-demand”. A snapshot can be scheduled based on the available options. Example: hourly/weekly
+
+- Select **Run Once > Yes, Continue** to trigger the snapshot. Backup operations convert application and volume snapshots into backups by transforming them into an infrastructure-independent format and then storing them in a target location (Google Cloud Bucket).
+
+![run-once](../../assets/run-once.png)
+
+The snapshots status can be monitored and exported from the Dashboard. The backup had been successfully completed and exported to the storage location. 
+
+### From Target Cluster
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:250]: Should this be 4 # and header changed to "Restore"?

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+Make sure that Replicated PV Mayastor has been installed, pools have been configured, and storageclasses are created (same as backup cluster) before restoring the target cluster. Refer to the [OpenEBS Installation Documentation](../../quickstart-guide/installation.md#installation-via-helm) for more details.
+
+Make sure that Kasten has been installed, volumesnapshotclass are created, and the dashboard is accessible before restoring the target cluster. Refer to the [Install Kasten section](#install-kasten) for more details.
+
+:::note
+The Location profile must be located in the exact same location as our backup, otherwise the restore would be unsuccessful. 
+:::
+
+The dashboard for the target cluster is now accessible since the backup was taken and as the above mentioned configurations were implemented on the restore cluster.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:260]: Not able to understand this sentence

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+![create-new-policy](../../assets/create-new-policy.png)
+
+#### Restore
+
+Once the policies are created, the import and restore processes can be initiated by selecting **Run Once**.  
+
+![run-once-restore](../../assets/run-once-restore.png)
+
+The backup had been successfully completed.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:284]: Is this correct?

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 2 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:108]: Not required, kasten document has some steps like how an user can achieve the dashboard UI, i have just followed that. user can either use port forwarding,nodePort or ingress

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+
+In this guide, we will utilize Kasten to create a backup of a sample Nginx application with a Replicated PV Mayastor from a cluster, transfer the backup to an object store, and restore it on a different cluster.
+
+## Requirements
+
+### Replicated PV Mayastor
+
+Replicated PV Mayastor, a high-performance, container-native storage that provides persistent storage for Kubernetes applications. It supports various storage types, including local disks, NVMe, and more. It integrates with Kubernetes and provides fast, efficient snapshots and creates snapshots at the storage layer, providing a point-in-time copy of the data. These snapshots are highly efficient and consume minimal space, as they only capture changes since the last snapshot.
+
+Make sure that Replicated PV Mayastor has been installed, pools have been configured, and applications have been deployed before proceeding to the next step. Refer to the [OpenEBS Installation Documentation](../../quickstart-guide/installation.md#installation-via-helm) for more details.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:23]: This is a generic statement.
Since we are planning to backup a test nginx application, we need to provide the yaml to install the application. Also, it's important that the volume mode is specified as 'filesystem'.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 3 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:248]: @balaharish7 rephrase this as in confluence. 

You can monitor the status of the snapshots and export from Dashboard. we could see the backup completed and exported to our storage location.

@avishnu We have an image attached which shows how this can be exported as part of backup steps.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 2 above)
Comment (avishnu) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:108]: That's what I meant - it should be 1 or 2, not 2 after 1.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 5 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:260]: @balaharish7 rephrase this again as mentioned in confluence. 

Now that we have taken the backup and followed the above configurations on the restore cluster, we have the dashboard available for the target cluster.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 2 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:108]: got it. @balaharish7 change this as user can follow either one of these steps.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 7 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:23]: @balaharish7 YAML added now to confluence, please update the doc

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 6 above)
Comment (balaharish7) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:284]: Changed "backup" to "restore".

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 7 above)
Comment (balaharish7) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:23]: Yaml added to the docs

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md (hunk 1 above)
Comment (balaharish7) [reply docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:5]: I've added the term "OpenEBS" in the keywords. Although, the search engine renders the documents even without the term OpenEBS.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 8]
```diff
@@ -32,15 +33,15 @@ Kasten K10, a Kubernetes-native data management platform that offers backup, dis
...
+As an example, we will be using `openebs-hostpath` storageclass as a global persistence storageclass for the Kasten installation. 
 
 1. Install Kasten.
 
 ```
 helm install k10 kasten/k10 --namespace=kasten-io --set global.persistence.storageClass=openebs-hostpath
 ```
 
-2. Once you have installed Kasten, verify that Kasten has installed correctly.
+2. Once you have installed Kasten, verify that Kasten has been installed correctly.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:44]: We don't need 'Once you have installed Kasten'
Start with 'Verify...'

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 9]
```diff
@@ -91,21 +92,23 @@ driver: io.openebs.csi-mayastor
...
 ### Validate Dashboard Access
 
-Use the following `kubectl` command to forward a local port to Kasten ingress port or change the 'svc' type from **ClusterIP** to **NodePort** to establish a connection to it.
+Use the following `kubectl` command to forward a local port to the Kasten ingress port or change the 'svc' type from **ClusterIP** to **NodePort** to establish a connection to it.
 
 :::note
 By default, the Kasten dashboard is not exposed externally.
 :::
 
 In this example we have changed the 'svc' type to **NodePort**:
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:101]: Line 101 is supposed to be for changing service type to Ndeport.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 10]
```diff
@@ -171,6 +174,57 @@ Make sure the service account has necessary permissions.
 
 In this example, We have deployed a sample Nginx test application for backup and restore.
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:175]: Nginx test application with a Mayastor PVC where volume mode is Filesystem.

File: docs/main/Solutioning/backup-and-restore/kasten-br-fs.md [hunk 11]
```diff
@@ -98,7 +98,7 @@ Use the following `kubectl` command to forward a local port to the Kasten ingres
 By default, the Kasten dashboard is not exposed externally.
 :::
 
-In this example we have changed the 'svc' type to **NodePort**:
+In this example we have changed the 'service type' type to **NodePort**:
```
Comment (avishnu) [inline docs/main/Solutioning/backup-and-restore/kasten-br-fs.md:101]: I meant this is for the 2nd approach, not the 1st IMO

**Review by avishnu — APPROVED** [review]
